Thing.h

```diff
@@ -724,43 +723,33 @@ class ThingDevice {
       type++;
     }
 
-    JsonArray links = descr.createNestedArray("links");
-    {
-      JsonObject links_prop = links.createNestedObject();
-      links_prop["rel"] = "properties";
-      links_prop["href"] = "/things/" + this->id + "/properties";
-    }
-
-    {
-      JsonObject links_prop = links.createNestedObject();
-      links_prop["rel"] = "actions";
-      links_prop["href"] = "/things/" + this->id + "/actions";
-    }
-
-    {
-      JsonObject links_prop = links.createNestedObject();
-      links_prop["rel"] = "events";
-      links_prop["href"] = "/things/" + this->id + "/events";
-    }
```

Comment:
In the gateway we've turned these links into forms using `queryallactions`, `subscribeallevents` and `unsubscribeallevents`.

I would at least suggest keeping the /actions endpoint using the `queryallactions` operation (which is currently loosely defined in the Thing Description specification).

Events are a bit trickier since there's not really a W3C compliant way to describe the current [Events](https://webthings.io/api/#events-resource) resource in the API which is more like an event log. For the gateway what we decided to do is to implement [implement Server-Sent Events](https://github.com/WebThingsIO/gateway/pull/2863) and expose those endpoints in the Thing Description instead.

I would suggest doing this in webthing-arduino too, but I think in the meantime it's reasonable to remove this. That does raise the question of how you're describing the [Event](https://webthings.io/api/#event-resource) resources though?

Reply:
Well, I removed it, because it made the application crash when querying certain things. I don't think the Arduino Library needs to be a feature complete library at the moment. The important requirement was to migrate it to the W3C WoT TD 1.0 standard. Which I completed.

Reply:
You didn't answer how you are describing the Event resources in the TD. I don't know of a W3C valid way of describing the current API, see https://github.com/w3c/wot-thing-description/issues/892